test/api/dapps.js

```diff
@@ -18,214 +22,427 @@ before(function (done) {
 	});
 });
 
-describe('GET /dapps', function () {
+before(function (done) {
+	var account = node.randomAccount();
+	var options = node.guestbookDapp;
+	var transaction = node.lisk.dapp.createDapp(node.gAccount.password, null, options);
+	node.guestbookDapp.transactionId = transaction.id;
 
-	before(function (done) {
-		node.onNewBlock(done);
-	});
+	postTransaction(transaction, done);
+});
 
-	function getDapps (params, done) {
-		node.get('/api/dapps?' + params, done);
+before(function (done) {
+	var account = node.randomAccount();
+	var options = node.blockDataDapp;
+	var transaction = node.lisk.dapp.createDapp(node.gAccount.password, null, options);
+	node.blockDataDapp.transactionId = transaction.id;
+
+	postTransaction(transaction, done);
+});
+
+before(function (done) {
```

Comment:
I did like this to separate fixture setup A from B. There is similar pattern in other tests.